### PR TITLE
feat: upgrade template structure

### LINKED ISSUE #92: Better template structure
Based on [this](https://github.com/ubiquity/ts-template/issues/67) discussion and [this](https://github.com/ubiquity/uusd.ubq.fi/pull/13) PR we agree that right now the https://github.com/ubiquity/ts-template lacks structure and complex UIs (like `pay.ubq.fi`) are turned into a mess.

Since we don't want to use a js framework it makes sense at least to structure https://github.com/ubiquity/ts-template properly.

As as part of this issue the https://github.com/ubiquity/ts-template should be modified this way:
1. Add 2 separate html pages ([example1](https://github.com/zugdev/uusd.ubq.fi/blob/f4c6623cec130cff75e299ca47f0a45b63ace819/static/home.html), [example2](https://github.com/zugdev/uusd.ubq.fi/blob/f4c6623cec130cff75e299ca47f0a45b63ace819/static/mint.html))
2. Add 2 separate js "controllers" for those html pages ([example1](https://github.com/zugdev/uusd.ubq.fi/blob/f4c6623cec130cff75e299ca47f0a45b63ace819/src/pages/home.ts), [example2](https://github.com/zugdev/uusd.ubq.fi/blob/f4c6623cec130cff75e299ca47f0a45b63ace819/src/pages/mint.ts))
3. Add 2 reusable js components ([example](https://github.com/zugdev/uusd.ubq.fi/blob/f4c6623cec130cff75e299ca47f0a45b63ace819/src/common/collateral.ts))
4. Add a router ([example](https://github.com/zugdev/uusd.ubq.fi/blob/f4c6623cec130cff75e299ca47f0a45b63ace819/src/router.ts))
5. Add global `on-load.ts` and `listeners.ts` files for easier review and state management (not sure how it's gonna look like and if it's worth impleneting it)

In the end we should get a project structure similar to the one from https://github.com/vercel/next.js where it should be clear where exactly to put UI components, pages, handlers, etc...

### PR DESCRIPTION
**Pull request by obeys**
Resolves #92

Changes: 

Added `src` which will include:
- two components
- four controllers

- a router
- and `on-load.ts` for global state initialization

`static` directory will now only include different html pages
- index (which will serve as a layout)
- home
- 404
- and two different example pages

QA:

- 

How to QA and setup:
- deploy to any web host

### CODE REVIEW

**Comment by obeys** [comment]
waiting on @zugdev 🧛

**Comment by zugdev** [comment]
this is ok, not considering style. if I were to consider then at least the 404 page should look decent at least because it can be shared between implementations. @0x4007 we should probably settle on a header style for our UIs

**Review by zugdev — APPROVED** [review]

File: src/controllers/404.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
+export async function load404Page() {
+  const contentArea = document.getElementById("content-area");
+
+  if (contentArea) {
+    try {
+      const response = await fetch("./404.html");
```
Comment (0x4007) [inline src/controllers/404.ts]: Why wouldn't you prefetch at build time aka import a string this doesn't make sense to me

File: src/controllers/home.ts [hunk 2]
```diff
@@ -0,0 +1,13 @@
+export async function loadHomePage() {
+  const contentArea = document.getElementById("content-area");
+
+  if (contentArea) {
+    try {
+      const response = await fetch("./home.html");
```
Comment (0x4007) [inline src/controllers/home.ts]: Same here

**Comment by obeys** [comment]
@0x4007 jest was failing because it was trying to parse `html` files as `jsx` which isn't supported. I tried adding a babel config file which is what it was suggesting me to do but jest was still failing.

I've converted both 404 and home page into components, let me know if there's anything else that I need to do.

**Comment by 0x4007** [comment]
I think linking a shared 404 page would be much more maintainable across all our UIs. For example dao.ubq.fi/404

**Comment by obeys** [comment]
> I think linking a shared 404 page would be much more maintainable across all our UIs. For example dao.ubq.fi/404

Where would i be able to find the SRC for that page by any chance? I tried searching the `ubiquity` org but was unable to find any `dao.ubq.fi` repos

**Comment by obeys** [comment]
Ok I have managed to somewhat recreate that 404 page, I feel like it looks a lot better now but lmk @0x4007 !

**Comment by 0x4007** [comment]
No use just a link to it

**Comment by obeys** [comment]
> No use just a link to it

You want me to redirect to `https://dao.ubq.fi/404` incase of a 404?

Or I can display it with an iframe if that's what you mean

**Comment by 0x4007** [comment]
Redirect is fine

**Review by Hassanalmotawe89 — APPROVED** [review]

**Comment by obeys** [comment]
changed 404 page logic to redirect and removed the component + styles.

**Review by 0x4007 — APPROVED** [review]

**Comment by obeys** [comment]
is this good to be merged? @rndquu

**Review by rndquu — APPROVED** [review]

**Pull request merged**